src/overlays/actors/ovl_En_Invadepoh_Demo/z_en_invadepoh_demo.c

```diff
@@ -54,115 +124,689 @@ static InitChainEntry D_80C1AA88[] = {
+
+    this->actionFunc = EnInvadepohDemo_Cow_WaitForObject;
+}
+
+void EnInvadepohDemo_Ufo_Init(EnInvadepohDemo* this, PlayState* play) {
+    Actor_ProcessInitChain(&this->actor, sUfoInitChain);
+    this->actionFunc = EnInvadepohDemo_SelectCueAction;
+    this->drawFlags |= DRAW_FLAG_SHOULD_DRAW;
+}
+
+void EnInvadepohDemo_CowTail_Init(EnInvadepohDemo* this, PlayState* play) {
+    Actor_ProcessInitChain(&this->actor, sCowTailInitChain);
+    this->objectIndex = Object_GetIndex(&play->objectCtx, OBJECT_COW);
+    if (this->objectIndex < 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->actionFunc = EnInvadepohDemo_CowTail_WaitForObject;
+}
+
+void EnInvadepohDemo_Cow_Destroy(EnInvadepohDemo* this, PlayState* play) {
+    Actor* thisx = &this->actor;
+
+    if (thisx->child != NULL) {
+        thisx->child->parent = NULL;
+    }
+
+    Actor_Kill(thisx->child);
+}
+
+void EnInvadepohDemo_CowTail_Destroy(EnInvadepohDemo* this, PlayState* play) {
+    if (this->actor.parent != NULL) {
+        this->actor.parent->child = NULL;
+    }
+}
+
+void EnInvadepohDemo_Alien_Idle(EnInvadepohDemo* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnInvadepohDemo_Alien_FollowPath(EnInvadepohDemo* this, PlayState* play) {
+    Path* path = &play->setupPathList[this->pathIndex];
+    s32 pathCount = path->count;
+    Vec3s* points = Lib_SegmentedToVirtual(path->points);
+    Vec3f point;
+
+    if (pathCount != 0) {
+        Math_Vec3s_ToVec3f(&point, &points[this->currentPointIndex]);
+        if (Math_Vec3f_StepTo(&this->actor.world.pos, &point, this->speed) < this->speed) {
+            this->currentPointIndex++;
+            if (this->currentPointIndex >= pathCount) {
+                Actor_Kill(&this->actor);
+            }
+        } else {
+            SkelAnime_Update(&this->skelAnime);
+        }
+    }
+}
+
+void EnInvadepohDemo_Romani_Idle(EnInvadepohDemo* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnInvadepohDemo_Romani_FollowPath(EnInvadepohDemo* this, PlayState* play) {
+    Path* path = &play->setupPathList[this->pathIndex];
+    s32 pathCount = path->count;
+    Vec3s* points = Lib_SegmentedToVirtual(path->points);
+    Vec3f point;
+
+    if (pathCount != 0) {
+        Math_Vec3s_ToVec3f(&point, &points[this->currentPointIndex]);
+        if (Math_Vec3f_StepTo(&this->actor.world.pos, &point, this->speed) < this->speed) {
+            this->currentPointIndex++;
+            if (this->currentPointIndex >= pathCount) {
+                Actor_Kill(&this->actor);
+            }
+        } else {
+            SkelAnime_Update(&this->skelAnime);
+        }
+    }
+}
+
+/**
+ * Positions the cow tail actor at the appropriate spot on the cow's body and rotates it to
+ * match the cow's rotation. This function is also responsible for playing the cow's animation.
+ */
+void EnInvadepohDemo_Cow_PositionTailAndPlayAnimation(EnInvadepohDemo* this, PlayState* play) {
```

Comment:
Forgot to post about it when you brought it up in discord. Maybe something like `EnInvadepohDemo_Cow_UpdateModel` or `EnInvadepohDemo_Cow_UpdateCommon` or `EnInvadepohDemo_Cow_CommonAction`?

Reply:
Went with `EnInvadepohDemo_Cow_UpdateCommon` here. I think I also said this on Discord, but EnInvadepoh pretty much does this exact thing in the Cow's Update function, so it feels fitting to call this "Update" too.